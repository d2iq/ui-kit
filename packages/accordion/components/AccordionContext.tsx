import React, { createContext, useState } from "react";

interface AccordionProviderProps {
  initialExpandedItems?: string[];
  allowMultipleExpanded?: boolean;
  onChange?: (expandedItems: string[]) => void;
}

type AccordionContext = {
  expandedItems: string[];
  setExpandedItems: (toggledItem: string, isExpanded?: boolean) => void;
};

export const Context = createContext<AccordionContext | null>(null);

export const Provider: React.FC<AccordionProviderProps> = ({
  allowMultipleExpanded,
  children,
  initialExpandedItems = [],
  onChange
}) => {
  const [expandedItems, setExpanded] = useState(initialExpandedItems);
  const setExpandedItems = (toggledItem, isExpanded) => {
    if (!isExpanded) {
      allowMultipleExpanded
        ? setExpanded([...expandedItems, toggledItem])
        : setExpanded([toggledItem]);
      return;
    }

    setExpanded(
      expandedItems.filter(expandedItem => expandedItem !== toggledItem)
    );
  };

  React.useEffect(() => {
    if (onChange) {
      onChange(expandedItems);
    }
  }, [expandedItems]);

  return (
    <Context.Provider
      value={{
        expandedItems,
        setExpandedItems
      }}
    >
      {children}
    </Context.Provider>
  );
};
