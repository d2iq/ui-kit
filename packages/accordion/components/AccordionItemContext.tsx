import React, { createContext } from "react";
import nextId from "react-id-generator";

interface AccordionProviderProps {
  id?: string;
  expandedItems: string[];
  children: React.ReactNode;
}

type AccordionItemContext = {
  baseId: string;
  contentId: string;
  headingId: string;
  isExpanded: boolean;
};

export const Context = createContext<AccordionItemContext | null>(null);

export const Provider = ({
  children,
  expandedItems,
  id: baseId = nextId("accordionProvider-")
}: AccordionProviderProps) => (
  <Context.Provider
    value={{
      baseId,
      contentId: `${baseId}-content`,
      headingId: `${baseId}-heading`,
      isExpanded: expandedItems.includes(baseId)
    }}
  >
    {children}
  </Context.Provider>
);
