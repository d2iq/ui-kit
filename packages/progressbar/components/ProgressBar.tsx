import * as React from "react";
import { cx } from "@emotion/css";
import { Flex, FlexItem } from "../../styleUtils/layout";
import { SpacingBox } from "../../styleUtils/modifiers";
import { Text, CaptionText } from "../../styleUtils/typography";
import { progressBar, progressBarStaged, progressBarFill } from "../style";
import { visuallyHidden } from "../../shared/styles/styleUtils";

export interface ProgressBarDatum {
  percentage: number;
  color?: React.CSSProperties["fill"];
}

export enum ProgressBarSizes {
  LARGE = "large",
  DEFAULT = "default"
}

export interface ProgressBarProps {
  /**
   * A very short description of the data being presented
   */
  caption?: React.ReactNode;
  /**
   * Objects that define the proportion and color of segments in the progress bar
   */
  data: ProgressBarDatum[];
  /**
   * Whether the data being represennted is still being calculated or is related to a task that is still in-flight
   */
  isProcessing?: boolean;
  /**
   * The primary piece of data being represented
   */
  value?: React.ReactNode;
  /**
   * Shows the progress bar and value on the same line, and hides the caption above the progress bar.
   */
  isCondensedLayout?: boolean;
  /**
   * Which size variant to use
   */
  size?: ProgressBarSizes;
}

const ProgressBar: React.StatelessComponent<ProgressBarProps> = ({
  caption,
  data,
  value,
  size,
  isProcessing,
  isCondensedLayout
}) => {
  const renderProgressBar = () => (
    <svg
      className={cx(progressBar(size), {
        [progressBarStaged]: isProcessing
      })}
      data-cy="progressBar-chart"
      role="presentation"
    >
      {data.map((datum, i) => {
        const { percentage, color } = datum;
        const precedingTotal = data
          .slice(0, i)
          .reduce((acc, curr) => acc + curr.percentage, 0);

        return (
          <rect
            x={`${precedingTotal}%`}
            width={`${percentage}%`}
            height="100%"
            className={progressBarFill(color)}
            key={`pbSegment-${i}`}
          />
        );
      })}
    </svg>
  );

  return isCondensedLayout ? (
    <Flex align="center" gutterSize="s">
      <FlexItem>{renderProgressBar()}</FlexItem>
      {caption ? (
        <FlexItem flex="shrink" className={visuallyHidden}>
          {caption}
        </FlexItem>
      ) : null}
      {value ? (
        <FlexItem flex="shrink">
          <Text tag="div">{value}</Text>
        </FlexItem>
      ) : null}
    </Flex>
  ) : (
    <React.Fragment>
      <SpacingBox side="bottom" spacingSize="xxs">
        <Flex align="center" justify="flex-end" gutterSize="s">
          {caption ? (
            <FlexItem>
              <CaptionText wrap="truncate" tag="div">
                {caption}
              </CaptionText>
            </FlexItem>
          ) : null}
          {value ? (
            <FlexItem flex="shrink">
              <Text weight="medium" tag="div">
                {value}
              </Text>
            </FlexItem>
          ) : null}
        </Flex>
      </SpacingBox>
      {renderProgressBar()}
    </React.Fragment>
  );
};

export default ProgressBar;
