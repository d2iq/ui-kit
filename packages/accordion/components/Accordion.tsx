import * as React from "react";
import { Stack } from "../../styleUtils/layout";
import { Provider as AccordionProvider } from "./AccordionContext";
import { AccordionBaseProps } from "../types";

interface AccordionProps extends AccordionBaseProps {
  /**
   * An array of open accordion panel IDs
   */
  initialExpandedItems?: string[];
  children: React.ReactNode[];
}

const Accordion = ({
  allowMultipleExpanded = false,
  "data-cy": dataCy = "accordion",
  children,
  initialExpandedItems,
  onChange
}: AccordionProps) => {
  return (
    <AccordionProvider
      allowMultipleExpanded={allowMultipleExpanded}
      initialExpandedItems={initialExpandedItems}
      onChange={onChange}
    >
      <Stack data-cy={dataCy}>{children}</Stack>
    </AccordionProvider>
  );
};

export default Accordion;
