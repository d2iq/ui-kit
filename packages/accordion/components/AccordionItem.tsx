import * as React from "react";
import { useId } from "react-id-generator";
import { Context as AccordionContext } from "./AccordionContext";
import { Provider as AccordionItemProvider } from "./AccordionItemContext";

export interface AccordionItemProps {
  /**
   * human-readable selector used for writing tests
   */
  ["data-cy"]?: string;
  /**
   * A custom ID for the accordion panel
   */
  id?: string;
}

const AccordionItem: React.FC<AccordionItemProps> = ({
  children,
  "data-cy": dataCy,
  id
}) => {
  const [generatedId] = useId(1, "accordionItem");
  const accordionItemId = id || generatedId;
  const accordionContext = React.useContext(AccordionContext);
  const isExpanded = accordionContext?.expandedItems.includes(accordionItemId);

  return (
    <AccordionItemProvider
      expandedItems={accordionContext?.expandedItems || []}
      id={accordionItemId}
    >
      <div
        data-cy={[dataCy, ...(isExpanded ? [`${dataCy}.expanded`] : [])].join(
          " "
        )}
      >
        {children}
      </div>
    </AccordionItemProvider>
  );
};

AccordionItem.defaultProps = {
  "data-cy": "accordionItem"
};

export default AccordionItem;
