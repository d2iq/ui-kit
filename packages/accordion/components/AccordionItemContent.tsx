import * as React from "react";
import { Context as AccordionItemContext } from "./AccordionItemContext";
import { cx } from "@emotion/css";
import {
  visuallyHidden,
  border,
  padding
} from "../../shared/styles/styleUtils";
import { getAllFocusableChildNodes } from "../../utilities/getFocusableChildNodes";
import { accordionItemContent } from "../style";
import { SpaceSize } from "../../shared/styles/styleUtils/modifiers/modifierUtils";

const AccordionItemContent: React.FC<{
  /**
   * human-readable selector used for writing tests
   */
  ["data-cy"]?: string;
  /**
   * the amount of space between the border and the content
   */
  paddingSize?: SpaceSize;
}> = ({ children, "data-cy": dataCy, paddingSize }) => {
  const contentRef = React.useRef<HTMLDivElement>(null);
  const accordionItemContext = React.useContext(AccordionItemContext);
  React.useEffect(() => {
    const allFocusable = contentRef.current
      ? getAllFocusableChildNodes(contentRef.current)
      : [];

    if (!accordionItemContext?.isExpanded) {
      allFocusable.forEach(el => {
        el.setAttribute("tabindex", "-1");
      });
      return;
    }

    allFocusable.forEach(el => {
      el.removeAttribute("tabindex");
    });
  }, [contentRef, accordionItemContext?.isExpanded]);

  return (
    <div
      aria-labelledby={accordionItemContext?.headingId}
      id={accordionItemContext?.contentId}
      className={cx(
        border("all"),
        padding("all", paddingSize),
        accordionItemContent,
        {
          [visuallyHidden]: !accordionItemContext?.isExpanded
        }
      )}
      ref={contentRef}
      hidden={!accordionItemContext?.isExpanded}
      data-cy={dataCy}
    >
      {children}
    </div>
  );
};

AccordionItemContent.defaultProps = {
  paddingSize: "m",
  "data-cy": "accordionItemContent"
};

export default AccordionItemContent;
