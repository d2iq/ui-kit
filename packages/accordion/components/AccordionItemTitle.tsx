import * as React from "react";
import { cx } from "@emotion/css";
import ResetButton from "../../button/components/ResetButton";
import { Context as AccordionItemContext } from "./AccordionItemContext";
import { Context as AccordionContext } from "./AccordionContext";
import { textWeight } from "../../shared/styles/styleUtils";
import { fillWidth } from "../style";
import { HeadingLevel, AccordionTitleAppearances } from "../types";
import { headingReset } from "../../shared/styles/styleUtils/resets/headingReset";
import AccordionItemTitleOuter from "./AccordionItemTitleOuter";
import AccordionItemTitleInner from "./AccordionItemTitleInner";

export interface AccordionItemTitleProps {
  /**
   * Changes visual design attributes to communicate the status of the accordion item
   */
  appearance?: AccordionTitleAppearances;
  /**
   * human-readable selector used for writing tests
   */
  ["data-cy"]?: string;
  /**
   * Whether the accordion item can be expanded
   */
  disabled?: boolean;
  /**
   * Priority of the heading. Numbers map to <h1> through <h6>
   */
  headingLevel?: HeadingLevel;
}

const AccordionItemTitle: React.FC<AccordionItemTitleProps> = ({
  appearance,
  children,
  "data-cy": dataCy,
  disabled,
  headingLevel
}) => {
  const HeadingTag: keyof React.ReactHTML = `h${headingLevel}` as
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6";
  const accordionItemContext = React.useContext(AccordionItemContext);
  const accordionContext = React.useContext(AccordionContext);
  const handleToggle = () => {
    if (accordionContext !== null) {
      accordionContext.setExpandedItems(
        accordionItemContext!.baseId,
        accordionItemContext!.isExpanded
      );
    }
  };

  return (
    <AccordionItemTitleOuter
      appearance={appearance}
      data-cy={[
        dataCy,
        ...(appearance && appearance !== "default"
          ? [`${dataCy}.${appearance}`]
          : []),
        ...(disabled ? [`${dataCy}.disabled`] : [])
      ].join(" ")}
      isExpanded={accordionItemContext!.isExpanded}
      disabled={disabled}
    >
      <HeadingTag
        id={accordionItemContext?.headingId || ""}
        className={cx(headingReset, textWeight("medium"))}
        data-cy={`${dataCy}-heading`}
      >
        <ResetButton
          disabled={disabled}
          aria-controls={accordionItemContext?.contentId || ""}
          aria-expanded="false"
          onClick={handleToggle}
          className={fillWidth}
          data-cy={`${dataCy}-button`}
        >
          <AccordionItemTitleInner
            isExpanded={accordionItemContext?.isExpanded}
          >
            {children}
          </AccordionItemTitleInner>
        </ResetButton>
      </HeadingTag>
    </AccordionItemTitleOuter>
  );
};

AccordionItemTitle.defaultProps = {
  headingLevel: 3,
  "data-cy": "accordionItemTitle"
};

export default AccordionItemTitle;
