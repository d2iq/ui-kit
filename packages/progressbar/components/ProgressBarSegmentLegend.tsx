import * as React from "react";
import { Flex, FlexItem } from "../../styleUtils/layout";
import { ProgressBarDatum } from "./ProgressBar";
import { legendDot } from "../style";

export interface ProgressBarProps {
  /**
   * Objects that define the size and color of segments in the progress bar
   */
  data: ProgressBarDatum[];
}

const LegendItem = ({ color, percentage }: ProgressBarDatum) => (
  <Flex align="center" gutterSize="xxs">
    <FlexItem flex="shrink">
      <div className={legendDot(color)} />
    </FlexItem>
    <FlexItem>{percentage}%</FlexItem>
  </Flex>
);

const ProgressBarSegmentLegend = ({ data }) => (
  <Flex align="center" gutterSize="s">
    {data.map((datum, i) => (
      <FlexItem flex="shrink" key={`pbSegmentLegend-${i}`}>
        <LegendItem {...datum} />
      </FlexItem>
    ))}
  </Flex>
);

export default ProgressBarSegmentLegend;
