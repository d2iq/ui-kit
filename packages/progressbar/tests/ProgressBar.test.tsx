import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "@emotion/jest";
import toJson from "enzyme-to-json";

expect.addSnapshotSerializer(createSerializer());

import { ProgressBar } from "../";
import { ProgressBarSizes } from "../components/ProgressBar";

describe("ProgressBar", () => {
  it("renders with one segment of data", () => {
    const component = shallow(<ProgressBar data={[{ percentage: 40 }]} />);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with multiple segments of data", () => {
    const component = shallow(
      <ProgressBar
        data={[
          { color: "green", percentage: 40 },
          { color: "yellow", percentage: 20 }
        ]}
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders all sizes", () => {
    Object.keys(ProgressBarSizes).forEach(size => {
      const component = shallow(
        <ProgressBar
          size={ProgressBarSizes[size]}
          isProcessing={true}
          data={[{ percentage: 40 }]}
        />
      );

      expect(toJson(component)).toMatchSnapshot();
    });
  });
  it("renders isProcessing", () => {
    const component = shallow(
      <ProgressBar isProcessing={true} data={[{ percentage: 40 }]} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with caption text and value text", () => {
    const component = shallow(
      <ProgressBar caption="Caption" value="40%" data={[{ percentage: 40 }]} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
