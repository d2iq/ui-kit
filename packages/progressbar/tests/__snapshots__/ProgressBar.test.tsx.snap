// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders all sizes 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-background-position: 0 0;
    background-position: 0 0;
  }

  100% {
    -webkit-background-position: 12px 0;
    background-position: 12px 0;
  }
}

.emotion-0 {
  background-color: rgba(218,221,226,0.65);
  border-radius: 999px;
  display: block;
  height: 12px;
  width: 100%;
  -webkit-animation: animation-0 0.6s linear infinite;
  animation: animation-0 0.6s linear infinite;
  background-image: linear-gradient(
      45deg,
      transparent 25%,
      rgba(0, 0, 0, 0.1) 25%,
      rgba(0, 0, 0, 0.1) 50%,
      transparent 50%,
      transparent 75%,
      rgba(0, 0, 0, 0.1) 75%,
      rgba(0, 0, 0, 0.1) 100%
    );
  -webkit-background-size: 12px 12px;
  background-size: 12px 12px;
}

.emotion-1 {
  fill: var(--themeBrandPrimary, #7D58FF);
}

<Fragment>
  <SpacingBox
    side="bottom"
    spacingSize="xxs"
    tag="div"
  >
    <Flex
      align="center"
      direction="row"
      gutterSize="s"
      justify="flex-end"
      wrap="nowrap"
    />
  </SpacingBox>
  <svg
    className="emotion-0"
    data-cy="progressBar-chart"
    role="presentation"
  >
    <rect
      className="emotion-1"
      height="100%"
      key="pbSegment-0"
      width="40%"
      x="0%"
    />
  </svg>
</Fragment>
`;

exports[`ProgressBar renders all sizes 2`] = `
@keyframes animation-0 {
  0% {
    -webkit-background-position: 0 0;
    background-position: 0 0;
  }

  100% {
    -webkit-background-position: 12px 0;
    background-position: 12px 0;
  }
}

.emotion-0 {
  background-color: rgba(218,221,226,0.65);
  border-radius: 999px;
  display: block;
  height: 6px;
  width: 100%;
  -webkit-animation: animation-0 0.6s linear infinite;
  animation: animation-0 0.6s linear infinite;
  background-image: linear-gradient(
      45deg,
      transparent 25%,
      rgba(0, 0, 0, 0.1) 25%,
      rgba(0, 0, 0, 0.1) 50%,
      transparent 50%,
      transparent 75%,
      rgba(0, 0, 0, 0.1) 75%,
      rgba(0, 0, 0, 0.1) 100%
    );
  -webkit-background-size: 12px 12px;
  background-size: 12px 12px;
}

.emotion-1 {
  fill: var(--themeBrandPrimary, #7D58FF);
}

<Fragment>
  <SpacingBox
    side="bottom"
    spacingSize="xxs"
    tag="div"
  >
    <Flex
      align="center"
      direction="row"
      gutterSize="s"
      justify="flex-end"
      wrap="nowrap"
    />
  </SpacingBox>
  <svg
    className="emotion-0"
    data-cy="progressBar-chart"
    role="presentation"
  >
    <rect
      className="emotion-1"
      height="100%"
      key="pbSegment-0"
      width="40%"
      x="0%"
    />
  </svg>
</Fragment>
`;

exports[`ProgressBar renders isProcessing 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-background-position: 0 0;
    background-position: 0 0;
  }

  100% {
    -webkit-background-position: 12px 0;
    background-position: 12px 0;
  }
}

.emotion-0 {
  background-color: rgba(218,221,226,0.65);
  border-radius: 999px;
  display: block;
  height: 6px;
  width: 100%;
  -webkit-animation: animation-0 0.6s linear infinite;
  animation: animation-0 0.6s linear infinite;
  background-image: linear-gradient(
      45deg,
      transparent 25%,
      rgba(0, 0, 0, 0.1) 25%,
      rgba(0, 0, 0, 0.1) 50%,
      transparent 50%,
      transparent 75%,
      rgba(0, 0, 0, 0.1) 75%,
      rgba(0, 0, 0, 0.1) 100%
    );
  -webkit-background-size: 12px 12px;
  background-size: 12px 12px;
}

.emotion-1 {
  fill: var(--themeBrandPrimary, #7D58FF);
}

<Fragment>
  <SpacingBox
    side="bottom"
    spacingSize="xxs"
    tag="div"
  >
    <Flex
      align="center"
      direction="row"
      gutterSize="s"
      justify="flex-end"
      wrap="nowrap"
    />
  </SpacingBox>
  <svg
    className="emotion-0"
    data-cy="progressBar-chart"
    role="presentation"
  >
    <rect
      className="emotion-1"
      height="100%"
      key="pbSegment-0"
      width="40%"
      x="0%"
    />
  </svg>
</Fragment>
`;

exports[`ProgressBar renders with caption text and value text 1`] = `
.emotion-0 {
  background-color: rgba(218,221,226,0.65);
  border-radius: 999px;
  display: block;
  height: 6px;
  width: 100%;
}

.emotion-1 {
  fill: var(--themeBrandPrimary, #7D58FF);
}

<Fragment>
  <SpacingBox
    side="bottom"
    spacingSize="xxs"
    tag="div"
  >
    <Flex
      align="center"
      direction="row"
      gutterSize="s"
      justify="flex-end"
      wrap="nowrap"
    >
      <FlexItem
        flex="grow"
      >
        <CaptionText
          align="inherit"
          tag="div"
          wrap="truncate"
        >
          Caption
        </CaptionText>
      </FlexItem>
      <FlexItem
        flex="shrink"
      >
        <Text
          align="inherit"
          color="var(--themeTextColorPrimary, #1B2029)"
          data-cy="text"
          size="m"
          tag="div"
          weight="medium"
          wrap="wrap"
        >
          40%
        </Text>
      </FlexItem>
    </Flex>
  </SpacingBox>
  <svg
    className="emotion-0"
    data-cy="progressBar-chart"
    role="presentation"
  >
    <rect
      className="emotion-1"
      height="100%"
      key="pbSegment-0"
      width="40%"
      x="0%"
    />
  </svg>
</Fragment>
`;

exports[`ProgressBar renders with multiple segments of data 1`] = `
.emotion-0 {
  background-color: rgba(218,221,226,0.65);
  border-radius: 999px;
  display: block;
  height: 6px;
  width: 100%;
}

.emotion-1 {
  fill: green;
}

.emotion-2 {
  fill: yellow;
}

<Fragment>
  <SpacingBox
    side="bottom"
    spacingSize="xxs"
    tag="div"
  >
    <Flex
      align="center"
      direction="row"
      gutterSize="s"
      justify="flex-end"
      wrap="nowrap"
    />
  </SpacingBox>
  <svg
    className="emotion-0"
    data-cy="progressBar-chart"
    role="presentation"
  >
    <rect
      className="emotion-1"
      height="100%"
      key="pbSegment-0"
      width="40%"
      x="0%"
    />
    <rect
      className="emotion-2"
      height="100%"
      key="pbSegment-1"
      width="20%"
      x="40%"
    />
  </svg>
</Fragment>
`;

exports[`ProgressBar renders with one segment of data 1`] = `
.emotion-0 {
  background-color: rgba(218,221,226,0.65);
  border-radius: 999px;
  display: block;
  height: 6px;
  width: 100%;
}

.emotion-1 {
  fill: var(--themeBrandPrimary, #7D58FF);
}

<Fragment>
  <SpacingBox
    side="bottom"
    spacingSize="xxs"
    tag="div"
  >
    <Flex
      align="center"
      direction="row"
      gutterSize="s"
      justify="flex-end"
      wrap="nowrap"
    />
  </SpacingBox>
  <svg
    className="emotion-0"
    data-cy="progressBar-chart"
    role="presentation"
  >
    <rect
      className="emotion-1"
      height="100%"
      key="pbSegment-0"
      width="40%"
      x="0%"
    />
  </svg>
</Fragment>
`;
